### PR TITLE
[C#] FasterLog v2 iteration fix

### PR DESCRIPTION
**Pull request by badrishc**

### CODE REVIEW

File: cs/src/core/FasterLog/FasterLog.cs [hunk 1]
```diff
@@ -120,12 +120,21 @@ public sealed class FasterLog : IDisposable
...
         public FasterLog(FasterLogSettings logSettings)
+            : this(logSettings, logSettings.TryRecoverLatest)
+        { }
+
+        /// <summary>
+        /// Create new log instance
+        /// </summary>
+        /// <param name="logSettings">Log settings</param>
+        /// <param name="syncRecover">Recover synchronously</param>
+        private FasterLog(FasterLogSettings logSettings, bool syncRecover)
```
Comment (tli2) [inline cs/src/core/FasterLog/FasterLog.cs:137]: why not have syncRecover = true to cover the earlier case as well?

File: cs/src/core/FasterLog/FasterLog.cs [hunk 2]
```diff
@@ -167,7 +176,8 @@ public FasterLog(FasterLogSettings logSettings)
             commitPolicy.OnAttached(this);
 
             tolerateDeviceFailure = logSettings.TolerateDeviceFailure;
-            if (logSettings.TryRecoverLatest)
+
+            if (syncRecover)
```
Comment (tli2) [inline cs/src/core/FasterLog/FasterLog.cs:180]: Need to check recover latest here as well?

If not, maybe we don't need to separate flags for recover latest and recoversync?

**Review by tli2 — APPROVED** [review]

File: cs/src/core/FasterLog/FasterLog.cs (hunk 2 above)
Comment (badrishc) [reply cs/src/core/FasterLog/FasterLog.cs:180]: The private constructor needed to differentiate itself from the public one, so I made it contingent on recoverSync. It does not look at log settings, instead depends on caller.

File: cs/src/core/FasterLog/FasterLog.cs (hunk 1 above)
Comment (badrishc) [reply cs/src/core/FasterLog/FasterLog.cs:137]: We want public constructor to be clean with only one argument, the settings.

File: cs/src/core/FasterLog/FasterLog.cs (hunk 2 above)
Comment (badrishc) [reply cs/src/core/FasterLog/FasterLog.cs:180]: The CreateAsync call invokes this private constructor with this argument false, and takes care of async recovery itself.

File: cs/src/core/FasterLog/FasterLog.cs (hunk 1 above)
Comment (tli2) [reply cs/src/core/FasterLog/FasterLog.cs:137]: Ok

**Pull request merged**